front/src/modules/trainschedule/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx

```diff
@@ -196,12 +205,15 @@ const PowerRestrictionsSelector = ({
             <IntervalsEditor
               additionalData={formattedPathElectrificationRanges}
               intervalType={INTERVAL_TYPES.SELECT}
-              data={powerRestrictionRanges}
+              data={useFormattedIntervals ? formattedIntervals : powerRestrictionRanges}
               defaultValue={NO_POWER_RESTRICTION}
               emptyValue={NO_POWER_RESTRICTION}
               operationalPoints={electrificationChangePoints}
               selectOptions={powerRestrictionOptions}
-              setData={editPowerRestrictionRanges}
+              setData={(data) => {
```

Comment:
Keep this once you have added the useEffect
```suggestion
              setData={editPowerRestrictionRanges}
```

Reply:
fixed